### PR TITLE
Updated homework specification for lesson2

### PR DESCRIPTION
**Pull request by martamatos**
Update the homework descriptions and added the base code (homework.js).

### CODE REVIEW

File: lesson2/homework.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+
+const addNewTask = function(id, title, description, created, updated, dueDate, statusID, userID) {
```
Comment (flemeur) [inline lesson2/homework.js]: We probably don't want to pass the id as an argument, but maybe we should add a comment stating that they should return the newly added task as an object (including the auto increment ID)

File: lesson2/README.md [hunk 2]
```diff
@@ -54,12 +54,14 @@ Also, please read the following pages that explains the ACID database model and
 
 ## Homework
 
-- Write the necessary SQL statements to complete the functionality of the node task application. This includes, at the minimum, the following functionality:
-    - Adding a new task
-    - Changing the title, due date, or status of a task
-    - Marking a task as completed
-    - Deleting a task
-    - Deleting a user
+Complete the `homework.js` script so that you have a functional node application. In particular, add the necessary code to the functions:
```
Comment (flemeur) [inline lesson2/README.md]: I think this description is a big improvement. It's much harder to get wrong :smile:

I'm wondering if we should add an extra function for finding a tasks by id. Then we could validate the homework by calling all the functions at the end of the file. Maybe I'm overthinking it...

File: lesson2/homework.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+const changeTaskDueDate = function(taskID, newDueDate) {
+  
+};
+
+const changeTaskStatus = function(taskID, newStatus) {
+  
+};
+
+const markTaskAsCompleted = function(taskID, newStatus) {
```
Comment (flemeur) [inline lesson2/homework.js]: I guess the function would always use the same status ("Done") so i don't think we need the newStatus argument

File: lesson2/homework.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  
+};
+
+const deleteTask = function(taskID, taskID) {
```
Comment (flemeur) [inline lesson2/homework.js]: One taskID argument is probably enough :smile:

File: lesson2/homework.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  
+};
+
+const deleteUser = function(taskID, userID) {
```
Comment (flemeur) [inline lesson2/homework.js]: Do we need the taskID here?

File: lesson2/homework.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  
+};
+
```
Comment (flemeur) [inline lesson2/homework.js:24]: Down here we could put the validating code that runs all the functions one by one. Fx:

```
// Add a new task
const newTask = addNewTask('Wash dishes', 'blah blah', ....);
// Verify that it got an ID
if (newTask.id < 1) {
    console.log('Expected newly added task to have an ID');
    process.exit(1);
}

// Find task by ID
const foundTask = findTaskByID(newTask.id);
if (!foundTask) {
    console.log('Could not find task by ID');
    process.exit(1);
}

// Change task title
const newTitle = 'This is a new task title';
changeTaskTitle(newTask.id, newTitle);

const updatedTask = findTaskByID(newTask.id);
if (updatedTask.title !== newTitle) {
    console.log('Expected task to have new title');
    process.exit(1);    
}

```

**Review by flemeur — CHANGES_REQUESTED** [review]

File: lesson2/homework.js (hunk 3 above)
Comment (martamatos) [reply lesson2/homework.js]: true!

File: lesson2/homework.js (hunk 5 above)
Comment (martamatos) [reply lesson2/homework.js]: no, i'm not sure what i was thinking yesterday xD

File: lesson2/homework.js (hunk 4 above)
Comment (martamatos) [reply lesson2/homework.js]: not sure... xD

File: lesson2/README.md (hunk 2 above)
Comment (martamatos) [reply lesson2/README.md]: that's not a bad idea at all...

**Pull request merged**